### ISSUE TITLE
Blueprint table is a keyboard trap

### ISSUE THREAD

**Issue by joshcarvel** [palantir/blueprint#3834]
#### Environment

- __Package version(s)__:  Blueprint table 3.8.1
- __Browser and OS versions__:  Chrome

If possible, link to a minimal repro (fork [this code sandbox](https://codesandbox.io/s/blueprint-sandbox-et9xy)): https://codesandbox.io/s/blueprint-sandbox-hiwri

#### Steps to reproduce

1. Use tab key to move focus into blueprint table

#### Actual behavior

It's not possible to move focus out of the table using the keyboard - as demonstrated in repo above, once focus is in the table you cannot tab to either button on the page

#### Expected behavior

You should be able to tab back to the previous or onto the next element on the page.
